### PR TITLE
No need to redefine scandir on Haiku

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

File: src/scandir.h [hunk 1]
```diff
@@ -44,7 +44,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define dirent direct
 #endif
 
-#if defined(__BEOS__) || defined(__HAIKU__) || (defined(__sun) && defined(__SVR4)) || defined(WIN32)
```
Comment (Cheezmeister) [inline src/scandir.h:47]: Seems alright but explain like I'm 5? Is `scandir` just a builtin on Haiku?

**Review by Cheezmeister — APPROVED** [review]

File: src/scandir.h (hunk 1 above)
Comment (Begasus) [reply src/scandir.h:47]: I can't explain exactly how it has been, in the past there was a problem using scandir, this patch has been laying around for quite some time (not a developer here, but I've been seeing this in the past and not needed now).

**Comment by Begasus** [comment]
PS, this doesn't fix a build with a current commit checkout, but I'll get back to that later.

**Comment by h01ger** [comment]
https://dev.haiku-os.org/ticket/3374 has been implemented, does this mean this pull request can be closed as non-actionable?

**Comment by Begasus** [comment]
Will have to do another check (has been a while since last one), but I think this is still needed, current situation redefines scandir, which shouldn't be needed anymore.
Will get back on this shortly.

**Comment by perepujal** [comment]
Hi, any updates on this?

**Comment by Begasus** [comment]
Whoops, that isn't shortly :) Let me do a testrun today (I hope to get around to it), sorry for the delay.

**Comment by perepujal** [comment]
So seems still needed, merging, thanks!

**Pull request merged**